lib/chef/provider/windows_task.rb

```diff
@@ -225,6 +229,14 @@ def action_disable
 
         private
 
+        # seprated command arguments from :command property
+        def set_command_and_arguments
+          arr = new_resource.command.split
+          new_resource.command = arr[0]
+          arr.shift
+          new_resource.command_arguments = arr.join(" ")
```

Comment:
@bmiller08 you are correct that command.split won't fix the issue if user input is as given by you the given regular expression only work if user passes command as "C:\Program Files\example\program.exe" -arg1 -arg2. But if user pass this as "C:\Program Files\example\program.exe" -arg1 -arg2" the regex given by you will also not work here. There are couple of scenarios which I need to verify to give exact fix. Current code fix at least not break the chef-client running with chef-client -W or any more parameters..

Reply:
The second example isn't properly quoted and should fail anyway. However, it looks like using scan _should_ work here.

For example:
```
arr = new_resource.command.scan(/"[^"]+"|\S+/)
new_resource.command = arr[0]
arr.shift
new_resource.command_arguments = arr.join(" ")
```
This will correctly return the command and the arguments in both examples. Then the task would likely fail on Windows for the extra quote in the arguments. This would be **expected** behavior and leaves the responsibility on the user to properly quote strings and not on how Chef handles the input. Let me know what you think.

Reply:
@bmiller08 in recipe